controllers/users.js

```diff
@@ -191,6 +192,30 @@ const getUsers = async (req, res) => {
       }
     }
 
+    const isDeparted = req.query.departed === "true";
+
+    if (isDeparted) {
+      if (!dev) {
+        return res.boom.notFound("Route not found");
+      }
+      try {
+        const result = await dataAccess.retrieveUsers({ query: req.query });
+        const departedUsers = await userService.getUsersWithIncompleteTasks(result.users);
```

Comment:
we should avoid tight coupling with query parameters (`req.query.departed`). we can pass transformed and validated query parameters into services or models for better decoupling

Reply:
The same pattern is followed in the getUsers function, so it’s best to refactor the entire function to prevent new developers from following this approach in the future. Given the broader impact, it would be better to track this as a separate enhancement. Let me know your thoughts.

Reply:
Have created this issue here - https://github.com/Real-Dev-Squad/website-backend/issues/2274